### ISSUE TITLE
ScanNet Mesh Evaluation

### ISSUE THREAD

**Issue by AoxiangFan** [nianticlabs/mvsanywhere#28]
Hello,

Thanks for the great work!

I have been trying to evaluate the model using the ScanNet Mesh Evaluation Benckmark. And it seems the file `/scripts/evaluation.py` performs the job.

However, when I try to run: `python scripts/evaluation.py incremental --checkpoint $CHECKPOINT --output-dir $OUTPUT`, it reports an error: ` No module named mvsanywhere.test_incremental`.

Could you please provide the code for `mvsanywhere.test_incremental`?

**Comment by JamieWatson683** [comment]
Hi @AoxiangFan thanks for your interest!

Ah yes that script is left over from DoubleTake - on which this repo is based. 

Instead if you want to eval on Scannet meshes, you can run the `test.py` as usual, with fusion turned on (`--run_fusion True --depth_fuser custom_open3d`) to dump the meshes to disk, then you can use the script `scripts/evals/mesh_eval.py` to evaluate the meshes. 

Let me know if that works for you!

**Comment by AoxiangFan** [comment]
Hi @JamieWatson683  thank you for your reply!

I managed to get the meshes, but for the evaluation code `scripts/evals/mesh_eval.py`, it requires the visibility volume files as in line 164:

```
visibility_volume_path = (
            Path(args.visibility_volume_path) / scene_id / f"{scene_id}_volume.npz"
        )
```

The `{scene_id}_volume.npz` files seem not provided, could you help me with that?

**Comment by JamieWatson683** [comment]
Hi @AoxiangFan - please see the [DoubleTake](https://github.com/nianticlabs/doubletake?tab=readme-ov-file#-testing-and-evaluation) repo for those!

**Comment by zxr888** [comment]
Currently, we are reproducing the results of the MVSAnywhere algorithm and need to replicate its Mesh Evaluation on the ScanNet dataset. We ran the test.py and mesh_eval.py files using the following commands, but neither the resulting mesh nor the scores are satisfactory. The reproduction results and environment configuration are provided below. Could there be any parameter settings that might be causing these results?

commands：
python src/mvsanywhere/test.py \
    --name HERO_MODEL \
    --output_base_path /media/ubuntu20/SSD_zs/dataset/eval_pose_depth/data/scannet_mesh/result_mvsanywhere_2/ \
    --config_file configs/models/mvsanywhere_model.yaml \
    --load_weights_from_checkpoint weights/mvsanywhere_hero.ckpt \
    --data_config configs/data/scannet/scannet_default_test.yaml \
    --num_workers 8 \
    --run_fusion \
    --depth_fuser ours \
    --batch_size 4 \
    --mask_pred_depth


#prediction_dir=/media/ubuntu20/SSD_zs/dataset/eval_pose_depth/data/scannet_mesh/result_mvsanywhere_2/HERO_MODEL/scannet/default/meshes/0.04_3.0_open3d_masked_color/
prediction_dir=/media/ubuntu20/SSD_zs/dataset/eval_pose_depth/data/scannet_mesh/result_mvsanywhere_2/HERO_MODEL/scannet/default/meshes/0.04_3.0_ours_masked

CUDA_VISIBLE_DEVICES=0 python scripts/evals/mesh_eval.py \
    --groundtruth_dir /media/ubuntu20/SSD_zs/dataset/eval_pose_depth/data/scannet_mesh/scannet_test/groundtruth \
    --prediction_dir $prediction_dir \
    --visibility_volume_path /media/ubuntu20/SSD_zs/dataset/eval_pose_depth/data/scannet_mesh/scannet_test_visibility_volume


environment configuration:
Package                       Version       Editable project location
----------------------------- ------------- ------------------------------
absl-py                       2.3.1
actionlib                     1.14.0
actionlib-tools               1.14.0
addict                        2.4.0
angles                        1.9.13
annotated-types               0.7.0
antialiased-cnns              0.3
anyio                         4.11.0
appdirs                       1.4.4
argon2-cffi                   25.1.0
argon2-cffi-bindings          25.1.0
arrow                         1.4.0
asttokens                     3.0.0
async-lru                     2.0.5
attrs                         25.4.0
av                            16.0.1
awscli                        1.42.66
babel                         2.17.0
base_local_planner            1.17.3
beautifulsoup4                4.14.2
bidict                        0.23.1
bleach                        6.3.0
blinker                       1.9.0
bondpy                        1.8.6
botocore                      1.40.66
Brotli                        1.2.0
cachetools                    6.2.1
camera-calibration            1.17.0
camera-calibration-parsers    1.12.0
catkin                        0.8.10
catkin_virtualenv             0.6.1
certifi                       2025.10.5
cffi                          2.0.0
charset-normalizer            3.4.4
click                         8.3.0
colorama                      0.4.6
colorlog                      6.10.1
comet_ml                      3.54.0
comm                          0.2.3
ConfigArgParse                1.7.1
configobj                     5.0.9
contourpy                     1.3.2
controller-manager            0.20.0
controller-manager-msgs       0.20.0
controller-manager-tests      0.20.0
cras-py-common                2.3.9
cryptography                  46.0.3
cv-bridge                     1.16.2
cycler                        0.12.1
dash                          3.2.0
dataclasses                   0.8
debugpy                       1.8.17
decorator                     4.4.2
defusedxml                    0.7.1
descartes                     1.1.0
diagnostic-analysis           1.11.0
diagnostic-common-diagnostics 1.11.0
diagnostic-updater            1.11.0
dill                          0.4.0
docstring_parser              0.17.0
docutils                      0.19
dulwich                       0.24.8
dynamic-reconfigure           1.7.3
easydict                      1.13
efficientnet_pytorch          0.7.1
einops                        0.8.1
embreex                       2.17.7.post7
everett                       3.1.0
exceptiongroup                1.3.0
executing                     2.2.1
fastjsonschema                2.21.2
filelock                      3.20.0
fire                          0.7.1
Flask                         3.1.2
fonttools                     4.60.1
fpsample                      0.3.3
fqdn                          1.5.1
fsspec                        2024.12.0
gazebo_plugins                2.9.2
gazebo_ros                    2.9.2
gdown                         5.2.0
gencpp                        0.7.0
geneus                        3.0.0
genlisp                       0.4.18
genmsg                        0.6.0
gennodejs                     2.0.2
genpy                         0.6.15
gitdb                         4.0.12
GitPython                     3.1.45
gmpy2                         2.2.1
grpcio                        1.76.0
gsplat                        1.0.0
h11                           0.16.0
h2                            4.3.0
h5py                          3.15.1
hpack                         4.1.0
httpcore                      1.0.9
httpx                         0.28.1
hyperframe                    6.1.0
idna                          3.11
image-geometry                1.16.2
image-transport-codecs        2.3.9
ImageIO                       2.37.2
imageio-ffmpeg                0.6.0
importlib_metadata            8.7.0
interactive-markers           1.12.0
iopath                        0.1.10
ipykernel                     7.1.0
ipython                       8.37.0
ipywidgets                    8.1.8
isoduration                   20.11.0
itsdangerous                  2.2.0
jaxtyping                     0.3.3
jedi                          0.19.2
Jinja2                        3.1.6
jmespath                      1.0.1
joblib                        1.5.2
joint-state-publisher         1.15.1
joint-state-publisher-gui     1.15.1
jsk_data                      2.2.12
jsk-recognition-msgs          1.2.17
jsk_topic_tools               2.2.12
json5                         0.12.1
jsonpointer                   3.0.0
jsonschema                    4.25.1
jsonschema-specifications     2025.9.1
jupyter_client                8.6.3
jupyter_core                  5.9.1
jupyter-events                0.12.0
jupyter-lsp                   2.3.0
jupyter_server                2.17.0
jupyter_server_terminals      0.5.3
jupyterlab                    4.4.10
jupyterlab_pygments           0.3.0
jupyterlab_server             2.28.0
jupyterlab_widgets            3.0.16
kdl-parser-py                 1.14.2
kiwisolver                    1.4.9
kornia                        0.6.7
lark                          1.3.1
laser_geometry                1.6.7
lazy_loader                   0.4
lightning                     2.2.5
lightning-utilities           0.15.2
loguru                        0.7.2
lxml                          6.0.2
manifold3d                    3.2.1
mapbox_earcut                 1.0.3
Markdown                      3.10
markdown-it-py                4.0.0
MarkupSafe                    3.0.3
matplotlib                    3.10.7
matplotlib-inline             0.2.1
mavros                        1.19.0
mdurl                         0.1.2
mediapy                       1.2.4
message-filters               1.17.4
mistune                       2.0.5
moveit-core                   1.1.14
moviepy                       1.0.3
mpmath                        1.3.0
msgpack                       1.1.2
msgpack-numpy                 0.4.8
multiprocess                  0.70.18
mvsanywhere                   0.1.0         /home/ubuntu20/zxr/mvsanywhere
narwhals                      2.10.2
nbclient                      0.10.2
nbconvert                     7.4.0
nbformat                      5.5.0
nerfacc                       0.5.2
nerfstudio                    1.1.2
nest-asyncio                  1.6.0
networkx                      3.4.2
ninja                         1.11.1.1
nodeenv                       1.9.1
notebook_shim                 0.2.4
numpy                         1.26.0
nuscenes-devkit               1.2.0
open3d                        0.16.0
opencv-python                 4.8.0.76
opencv-python-headless        4.9.0.80
overrides                     7.7.0
packaging                     24.2
pandas                        2.3.3
pandocfilters                 1.5.1
parameterized                 0.9.0
parso                         0.8.5
pathos                        0.3.4
pexpect                       4.9.0
Pillow                        9.5.0
pillow_heif                   0.16.0
pip                           25.2
platformdirs                  4.5.0
plotly                        6.4.0
portalocker                   3.2.0
pox                           0.3.6
ppft                          1.7.7
pr2_controller_manager        1.8.21
proglog                       0.1.12
prometheus_client             0.23.1
prompt_toolkit                3.0.52
protobuf                      3.20.3
psutil                        7.1.3
ptyprocess                    0.7.0
pure_eval                     0.2.3
pyasn1                        0.6.1
pycocotools                   2.0.10
pycollada                     0.9.2
pycparser                     2.22
pydantic                      2.12.3
pydantic_core                 2.41.4
Pygments                      2.19.2
pyliblzfse                    0.4.1
pymeshlab                     2023.12.post3
pyngrok                       7.4.1
pyparsing                     3.2.5
pypng                         0.20220715.0
pyquaternion                  0.9.9
PySocks                       1.7.1
python-box                    6.1.0
python-dateutil               2.9.0.post0
python-engineio               4.12.3
python-json-logger            4.0.0
python-qt-binding             0.4.4
python-socketio               5.14.3
pytoml                        0.1.21
pytorch-lightning             2.5.5
pytorch-msssim                1.0.0
pytorch3d                     0.7.8
pytz                          2025.2
PyYAML                        6.0.3
pyzmq                         27.1.0
qt-dotgraph                   0.4.2
qt-gui                        0.4.2
qt-gui-app                    0.4.2
qt-gui-cpp                    0.4.2
qt-gui-py-common              0.4.2
rawpy                         0.25.1
referencing                   0.37.0
requests                      2.32.5
requests-toolbelt             1.0.0
resource_retriever            1.12.7
retrying                      1.4.2
rfc3339-validator             0.1.4
rfc3986-validator             0.1.1
rfc3987-syntax                1.1.0
rich                          14.2.0
robot_upstart                 0.4.2
RobotRaconteur                1.2.2
ros-introspection             1.2.1
ros_numpy                     0.0.5
ros_pytest                    0.2.1
ros_speech_recognition        2.1.28
rosapi                        0.11.17
rosbag                        1.16.0
rosbag_pandas                 0.5.4
rosboost-cfg                  1.15.8
rosbridge-library             0.11.17
rosbridge-server              0.11.17
rosclean                      1.15.8
roscompile                    1.2.1
roscreate                     1.15.8
rosdiagnostic                 1.11.0
rosdoc-lite                   0.2.11
rosgraph                      1.16.0
roslaunch                     1.16.0
roslib                        1.15.8
roslint                       0.12.0
roslz4                        1.16.0
rosmake                       1.15.8
rosmaster                     1.16.0
rosmsg                        1.16.0
rosnode                       1.16.0
rosparam                      1.16.0
rospilot                      1.6.1
rospy                         1.16.0
rospy_message_converter       0.5.9
rosserial_arduino             0.9.2
rosserial_client              0.9.2
rosserial_python              0.9.2
rosserial_vex_cortex          0.9.2
rosserial_vex_v5              0.9.2
rosserial_xbee                0.9.2
rosservice                    1.16.0
rostest                       1.16.0
rostopic                      1.16.0
rostwitter                    2.1.28
rosunit                       1.15.8
roswtf                        1.16.0
roswww                        0.1.13
rpds-py                       0.28.0
rqt_action                    0.4.9
rqt_bag                       0.5.1
rqt_bag_plugins               0.5.1
rqt-console                   0.4.12
rqt_dep                       0.4.12
rqt_graph                     0.4.14
rqt_gui                       0.5.3
rqt_gui_py                    0.5.3
rqt-image-view                0.4.17
rqt_launch                    0.4.9
rqt-logger-level              0.4.12
rqt-moveit                    0.5.11
rqt_msg                       0.4.10
rqt_nav_view                  0.5.7
rqt_plot                      0.4.13
rqt_pose_view                 0.5.11
rqt_publisher                 0.4.10
rqt_py_common                 0.5.3
rqt_py_console                0.4.10
rqt-reconfigure               0.5.5
rqt-robot-dashboard           0.5.8
rqt-robot-monitor             0.5.15
rqt_robot_steering            0.5.12
rqt-runtime-monitor           0.5.10
rqt-rviz                      0.7.0
rqt_service_caller            0.4.10
rqt_shell                     0.4.11
rqt_srv                       0.4.9
rqt-tf-tree                   0.6.4
rqt_top                       0.4.10
rqt_topic                     0.4.13
rqt_web                       0.4.10
rsa                           4.7.2
rtree                         1.4.1
rviz                          1.14.25
s3transfer                    0.14.0
scikit-image                  0.25.2
scikit-learn                  1.7.2
scipy                         1.15.3
semantic-version              2.10.0
Send2Trash                    1.8.3
sensor-msgs                   1.13.1
sentry-sdk                    2.43.0
setuptools                    69.5.1
shapely                       2.0.7
shtab                         1.7.2
simple-websocket              1.1.0
simplejson                    3.20.2
six                           1.17.0
smach                         2.5.2
smach-ros                     2.5.2
smach-viewer                  4.1.0
smclib                        1.8.6
smmap                         5.0.2
sniffio                       1.3.1
sound-play                    0.3.18
soupsieve                     2.8
splines                       0.3.0
srdfdom                       0.6.4
stack-data                    0.6.3
svg.path                      7.0
sympy                         1.14.0
tensorboard                   2.20.0
tensorboard-data-server       0.7.2
termcolor                     3.2.0
terminado                     0.18.1
tf                            1.13.4
tf-conversions                1.13.2
tf2-client                    1.0.0
tf2-geometry-msgs             0.7.7
tf2-kdl                       0.7.7
tf2-py                        0.7.7
tf2-ros                       0.7.7
tf2-sensor-msgs               0.7.7
tf2-server                    1.1.3
threadpoolctl                 3.6.0
tifffile                      2025.5.10
timm                          0.6.7
tinycss2                      1.4.0
tomli                         2.3.0
topic-tools                   1.16.0
torch                         2.1.2
torch-fidelity                0.3.0
torchmetrics                  1.8.2
torchvision                   0.16.2
tornado                       6.5.2
tqdm                          4.67.1
traitlets                     5.14.3
transforms3d                  0.4.2
trimesh                       4.9.0
triton                        2.1.0
typeguard                     4.4.4
typing_extensions             4.15.0
typing-inspection             0.4.2
tyro                          0.9.35
tzdata                        2025.2
urdfdom-py                    0.4.6
uri-template                  1.3.0
urllib3                       2.5.0
vhacdx                        0.0.9
viser                         0.1.27
wadler_lindig                 0.1.7
wandb                         0.22.3
wcwidth                       0.2.14
webcolors                     25.10.0
webencodings                  0.5.1
websocket-client              1.9.0
websockets                    15.0.1
Werkzeug                      3.1.3
wheel                         0.45.1
widgetsnbextension            4.0.15
wrapt                         2.0.0
wsproto                       1.2.0
wurlitzer                     3.1.1
xacro                         1.14.18
xatlas                        0.0.11
xxhash                        3.6.0
yourdfpy                      0.0.58
zipp                          3.23.0
zstandard                     0.23.0



scores:
"overall": {
        "acc": 0.2967336668074131,
        "compl": 0.2571480093896389,
        "chamfer": 0.276940838098526,
        "prc": 0.1318490044772625,
        "rec": 0.15707602601498366,
        "f1_score": 0.1415051597506353
    }
